### PR TITLE
Add ReductionParams to python frontend

### PR DESCRIPTION
**Pull request by rdspring1**
This PR add support for `ReductionParams` to the python frontend. All parameters are exposed except `std::vector<TensorView*> smem_persistent_buffers` and `CompileParams`.

### CODE REVIEW

**Comment by rdspring1** [comment]
!test

File: csrc/python_frontend/python_bindings.cpp [hunk 1]
```diff
@@ -686,6 +686,48 @@ void defineHeuristicParamBindings(py::module& nvfuser) {
       .PARAM(PointwiseParams, unroll_factor_outer)
       .PARAM(PointwiseParams, lparams);
 
+  // Reduction scheduler parameters
+  INITHEURISTICPARAMS(ReductionParams)
+      .PARAM(ReductionParams, fastest_dim)
```
Comment (jacobhinkle) [inline csrc/python_frontend/python_bindings.cpp:691]: Would it improve readabiity to sort these lines?

**Review by jacobhinkle — APPROVED** [review]

File: csrc/python_frontend/python_bindings.cpp (hunk 1 above)
Comment (rdspring1) [reply csrc/python_frontend/python_bindings.cpp:691]: I added them in order they appear in `ReductionParams`, but I thought alphabetical order is easier to parser.

**Comment by liqiangxl** [comment]
why `CompileParams` is skipped? When generate heuristic, outer reduction and all persistent schedulders will set the `maxrregcount` in compiler params.

**Comment by rdspring1** [comment]
> why CompileParams is skipped?

The python class is a reference to the original CPP class. It exists but just isn't exposed to the user. It is to be added later.

**Comment by rdspring1** [comment]
!test

**Pull request merged**